compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2707,6 +2707,14 @@ impl<'a> Parser<'a> {
                     );
                     err.emit();
                     this.bump();
+                } else if matches!(
+                    (&this.prev_token.kind, &this.token.kind),
+                    (token::DotDotEq, token::Gt)
+                ) {
+                    // `error_inclusive_range_match_arrow` handles cases like `0..=> {}`,
+                    // so we supress the error here
+                    err.cancel();
```

Comment:
can you instead make this `err.delay_as_bug()`? That ensures that if this line is ever reached and no other error is emitted, the compile won't succeed. It's a failsafe against future regressions.